doc/design/gan_api.md

```diff
@@ -0,0 +1,134 @@
+'''
+GAN implementation, just a demo.
+'''
+# pd for short, should be more concise.
+from paddle.v2 as pd
+import numpy as np
+import logging
+
+X = pd.data(pd.float_vector(784))
+
+# Conditional-GAN should be a class. 
+### Class member function: the initializer.
+class DCGAN(object):
```

Comment:
no need to be a class, simply 3 functions should be more simple.

Reply:
We may have a discussion later. I refer to several GAN models on github, pretty much all looks like this.